test/unit/src/common/actions/t_snap-builds.js

```diff
@@ -233,21 +223,18 @@ describe('snap builds actions', () => {
           });
       });
 
-      it('should dispatch error action', () => {
-        return store.dispatch(fetchSnap(repositoryUrl))
-          .then(() => {
-            expect(store.getActions()).toHaveActionOfType(
-              ActionTypes.FETCH_BUILDS_ERROR
-            );
-          });
+      it('should dispatch error action', async () => {
+        await store.dispatch(fetchSnap(repositoryUrl));
+        expect(store.getActions()).toHaveActionOfType(
+          ActionTypes.FETCH_BUILDS_ERROR
+        );
       });
 
-      it('should pass error message from repsponse to action', () => {
-        return store.dispatch(fetchSnap(repositoryUrl))
-          .then(() => {
-            const action = store.getActions().filter(a => a.type === ActionTypes.FETCH_BUILDS_ERROR)[0];
-            expect(action.payload.error.message).toBe('Bad repo URL');
-          });
+      it('should pass error message from repsponse to action', async () => {
+        await store.dispatch(fetchSnap(repositoryUrl));
+        const action = store.getActions().filter(
+          a => a.type === ActionTypes.FETCH_BUILDS_ERROR)[0];
```

Comment:
As much as I like expressiveness of arrow functions, I hate parsing statements like this.

Adding some brackets `(a) => a.type === ActionTypes.FETCH_BUILDS_ERROR` doesn't seem to help a lot, but I think moving `.filter` down one line make is a bit more readable:

```js
const action = store.getActions()
  .filter((a) => a.type === ActionTypes.FETCH_BUILDS_ERROR)[0];
```